homeassistant/components/seventeentrack/__init__.py

```diff
@@ -1 +1,194 @@
 """The seventeentrack component."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from seventeentrack import Client as SeventeenTrackClient
+from seventeentrack.errors import (
+    InvalidTrackingNumberError,
+    RequestError,
+    SeventeenTrackError,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE_ID, CONF_FRIENDLY_NAME, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.device_registry import DeviceRegistry
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .config_flow import get_client
+from .const import (
+    CONF_CARRIER_NAME,
+    CONF_SHOW_ARCHIVED,
+    CONF_TRACKING_NUMBER,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_SHOW_ARCHIVED,
+    DOMAIN,
+    PLATFORMS,
+    SERVICE_ADD_PACKAGE,
+)
+from .errors import AuthenticationError
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SERVICE_ADD_PACKAGE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE_ID): cv.string,
+        vol.Required(CONF_TRACKING_NUMBER): cv.string,
+        vol.Optional(CONF_CARRIER_NAME): cv.string,
+        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+    }
+)
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up the SeventeenTrack component."""
+    try:
+        client = await get_client(hass, config_entry.data)
+
+    except AuthenticationError as err:
+        raise ConfigEntryAuthFailed from err
+    except SeventeenTrackError as err:
+        raise ConfigEntryNotReady(
+            f"There was an error while logging in: {err}"
+        ) from err
+
+    coordinator = SeventeenTrackDataCoordinator(hass, config_entry, client)
+
+    config_entry.add_update_listener(coordinator.async_options_updated)
+    await coordinator.async_config_entry_first_refresh()
+    await coordinator.async_register_service()
+
+    hass.data.setdefault(DOMAIN, {})[config_entry.entry_id] = coordinator
+
+    hass.config_entries.async_setup_platforms(config_entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Unload SeventeenTrack Entry from config_entry."""
+
+    if unload_ok := await hass.config_entries.async_unload_platforms(
+        config_entry, PLATFORMS
+    ):
+        hass.data[DOMAIN].pop(config_entry.entry_id)
+        if not hass.data[DOMAIN]:
```

Comment:
Please check for the last loaded config entry instead.

Example:
https://github.com/home-assistant/core/blob/6f92c5381da822dd096dc61751fa0ff49efd0816/homeassistant/components/simplisafe/__init__.py#L463-L472